projects/timer_challenge/src/main.c

```diff
@@ -0,0 +1,43 @@
+#include <stdint.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define TIMER_PERIOD 500  // timer in ms
+
+typedef struct Counters {  // stores counters
+  uint8_t counter_a;
+  uint8_t counter_b;
+} Counters;
+
+void prv_timer_callback(SoftTimerId timer_id, void *context) {
```

Comment:
This is a private function, so it should be `static` so that it's local to the file; otherwise if some other file tried to define a `prv_timer_callback` there might be an issue.